server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/FilterToSearchQuery.java

```diff
@@ -75,9 +79,38 @@ private SearchQuery convertCondition(FilterCondition filter) {
         filter.getMaxSize().ifPresent(maxSize -> searchQuery.andCriteria(SearchQuery.sizeLessThan(maxSize)));
         filter.getMinSize().ifPresent(minSize -> searchQuery.andCriteria(SearchQuery.sizeGreaterThan(minSize)));
         filter.getHasAttachment().ifPresent(hasAttachment -> searchQuery.andCriteria(SearchQuery.hasAttachment(hasAttachment)));
+        filter.getHasKeyword().ifPresent(keywords -> {
+            keywords.stream()
+                .filter(hasKeyword -> !JmapFlag.UNSUPPORTED_KEYWORDS.contains(hasKeyword))
+                .flatMap(jmapFlag -> Guavate.stream(getFlagCriterion(jmapFlag, true)))
+                .peek(flagCriterion -> searchQuery.andCriteria(flagCriterion))
```

Comment:
The classic way to do this is with a **map** **reduce**. This side effect is a hack. Please fix it.